src/main/java/com/google/api/codegen/discogapic/transformer/DiscoGapicNamer.java

```diff
@@ -95,12 +100,26 @@ public static Name methodAsName(Method method) {
     return function.join(resource);
   }
 
-  /** Return the name of the resource from a given method's path. */
-  public static Name getResourceIdentifier(Method method) {
+  /** Return the name of the qualified resource from a given method's path. */
+  public static Name getQualifiedResourceIdentifier(Method method, String qualifyingResource) {
+    String methodPath = method.flatPath();
+    Name qualifier = Name.anyCamel(qualifyingResource);
+    String baseResource = getResourceIdentifier(methodPath).toLowerCamel();
+    Name baseResourceName = Name.anyCamel(baseResource);
```

Comment:
It seems like you could just initialize `baseResourceName` to the output of `getResourceIdentifier(methodPath)`

Reply:
Done.